app/controllers/patients_controller.rb

```diff
@@ -8,6 +8,11 @@ def index
 
     def show
         @patient = Patient.find(params[:id])
+        if @patient['doctors_id'].nil?
```

Comment:
I don't think you need to lookup doctor using its id. ActiveRecord should be populating the @patient.doctor field for you.